### PR TITLE
feat: add populate db

### PR DESCRIPTION
**Pull request by BMarink512**
### 🛠 Changes being made

- Add populate method at the endpoint db/populate; the method fetches the scraped data and inserts it into the DB. The method takes 3-4 minutes to run and finish

- Add base model schemas

- Add findOrCreate service on author, price, category, book, platform, and price

### 🏎 Quality check

- [X] Are your changes following SOLID principles?

- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

- [X] Walk away, take a break, re-read what you filled out above does it make sense if you were coming in cold? What extra context could you provide?

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -13,19 +13,107 @@ datasource db {
...
+  createdAt       DateTime  @default(now())
+  updatedAt       DateTime  @updatedAt
+  isDeleted       Boolean   @default(false)
+
+  author     Author     @relation(fields: [authorId], references: [id])
+  categories Category[]
+
+  prices    Price[]
+  favorites Favorite[]
+  raitings  Rating[]
```
Comment (riadeane) [inline prisma/schema.prisma]: dice 'raitings' y no 'ratings'

File: src/db/db.service.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      authorId: author.id,
+      categories: categories.map((category) => category.name),
+    })
+
+    await this.priceService.findOrCreate({
+      bookId: book.id,
+      platformId: platformId,
+      price: parseFloat(data.price.replace('$', '')),
+      date: new Date(),
+      productUrl: data.link,
```
Comment (riadeane) [inline src/db/db.service.ts:109]: una pregunta, esto efectivamente actualiza los precios cuando tenemos un precio para un book y platform que ya existe?

**Review by riadeane — CHANGES_REQUESTED** [review]

**Comment by riadeane** [comment]
Buen PR!! Solo ese typo y la duda que tengo

File: src/db/db.service.ts (hunk 2 above)
Comment (BMarink512) [reply src/db/db.service.ts:109]: actualmente no esta implementado, pero al metodo del servicio seria agregar un update dentro de la llamada al upsert donde se actualice el precio y la fecha, lo agrego ahora

**Review by riadeane — APPROVED** [review]
Super!!

**Comment by AlfreMK** [comment]
Please remember to do squash and merge to merge it

**Pull request merged**